azerothcore/azerothcore-wotlk
# fix(db): Translation to Ello Quest - Add Stitches Completion Event

In classic Wow after the Translation to Ello quest is completed an map-wide
event would commence.  This is the event where the Stitches creature
would make a run from across the area map from Raven Hill to Darkshire, eliminating the
Night Watchers/Players in its path.

Currently the area event does not exist and does not trigger.  

Fix includes data from https://github.com/TrinityCore/TrinityCore/issues/22369 and Azerothcore #68





## Changes Proposed:
Creation of the Stitches Event which is comprised of:
- New dialogue for Town Crier, Lord Ello Ebonlocke, Althea Ebonlocke, Stitches, and Watcher Cutford (Blizzlike)
- Broadcasts made to players of the map on status of event (Blizzlike)
- Content for Night Watch Camp (Blizzlike)
- Content for the Rotting Orchard entrance (Blizzlike)
- Content for Darkshire defense (Blizzlike)
- Stiches run across the map (Blizzlike)
- Watcher Cutford  run across the map (Blizzlike)
- Different NPC reactions and resets upon Stitches death at different points on the map 
- Stopping the event (Town Crier officially stops event but event will also stop after 30 minutes)
- Repositioned Night Watchers into more Blizzlike locations and orientations.
- Watcher Petras now patrols like he should (Blizzlike)
- Removed level 65 generic Night Watchers (non-Blizzlike)
- Remove old Watcher Cutford GUID.  Now SAI manages Watcher Cutford.


## Issues Addressed:
- Fixes #68 



## Tests Performed:
-  Completed Quest 252: Translation to Ello
Watch for Watcher Cutford
![WoWScrnShot_020821_192557](https://user-images.githubusercontent.com/65646022/107420242-ea838c00-6ad5-11eb-9668-8255db130433.jpg)

He talks with Lord Ello and then runs out and talks with Althea.  He then begins his run all the way to the Night Watch camp as Stitches is spawned and begins his run to Darkshire. 

Watcher Corwin spawn only for event
![WoWScrnShot_020821_192828](https://user-images.githubusercontent.com/65646022/107420966-b2c91400-6ad6-11eb-9f66-89a70dcd6a9c.jpg)

Watcher Sarys spawn only for event
![WoWScrnShot_020821_191602](https://user-images.githubusercontent.com/65646022/107420160-d3dd3500-6ad5-11eb-85e6-e2fb4d268689.jpg)

Night Watch Camp (Watcher Blomberg and Watcher Hutchins event only watchers) gets stomped as Cutford arrives.  He does not engage but runs into camp (Blizzlike).   Watcher Dodds and Paige respawn after event.
![WoWScrnShot_020821_191126](https://user-images.githubusercontent.com/65646022/107421249-f4f25580-6ad6-11eb-90e6-b80f4088b71b.jpg)

Stitches move to the Rotting Orchard path entrance (Watchers Thayer, Gelwin, Selkin, Merant event only watchers) and stomps those four night watchers after they move into position
![WoWScrnShot_020821_191252](https://user-images.githubusercontent.com/65646022/107421359-1c492280-6ad7-11eb-90da-0556400f8f7d.jpg)

Night Watchers form in Darkshire and engage Stitches ending the event (Watchers respwn after event except for Jordan and Keefer who follow normal respawn rules.)
![WoWScrnShot_020821_191535](https://user-images.githubusercontent.com/65646022/107421571-5c100a00-6ad7-11eb-8bb1-bc84b92a501b.jpg)




 - Merges without issue with AzerothCore rev. 71b40ed8bc75+ 2021-02-08 14:52:03 +0000 (master branch)
- Checked GUIDs in game_event, Smart_Scripts, Creature_Templete, Creature, Creature_Addon, Waypoints, and Creature_Texts  

## How to Test the Changes:

- Go to Darkshire
- .quest remove 252
- .quest add 252
- Turn in quest to Lord Ello Ebonlocke
- Event should start with Watcher Cutford running in a few seconds later. 


## Known Issues and TODO List:

- [ Night Watcher formations or makeup may not be 100% accurate]



## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
